src/character_body.cpp

```diff
@@ -1124,3 +1136,123 @@ bodypart_id Character::body_window( const std::string &menu_header,
     }
 }
 
+
+float Character::get_heartrate_index() const
+{
+    return heart_rate_index;
+}
+
+void Character::update_heartrate_index()
+{
+    // The following code was adapted from the heartrate function, which will now probably need to be rewritten to be based on the heartrate index.
+
+    //COLDBLOOD dependencies, works almost same way as temperature effect for speed.
+    const int player_local_temp = get_weather().get_temperature( pos() );
+    float temperature_modifier = 0.0f;
+    if( has_trait( trait_COLDBLOOD ) ) {
+        temperature_modifier = 0.002f;
+    }
+    if( has_trait( trait_COLDBLOOD2 ) ) {
+        temperature_modifier = 0.00333f;
+    }
+    if( has_trait( trait_COLDBLOOD3 ) || has_trait( trait_COLDBLOOD4 ) ) {
+        temperature_modifier = 0.005f;
+    }
+    const float hr_temp_mod = ( player_local_temp - 65 ) * temperature_modifier;
+    const float stamina_level = static_cast<float>( get_stamina() ) / get_stamina_max();
+    float hr_stamina_mod = 0.0f;
+    // The influence of stamina on heartrate seemeed excessive and was toned down.
+    if( stamina_level >= 0.9f ) {
```

Comment:
I don't like this as a nest of ifs. Let's just use an equation, the relationship can be mostly linear. I'm going to bed but I can think one up while I try to sleep.

Reply:
```
hr_stamina_mod = 1.6 * ( 1 - stamina_level)
```
This creates more or less the same formula  you have here.